### PR TITLE
 Doc updates for pyproject.toml and instructions for pypi

### PR DESCRIPTION
**Pull request by drewoldag**
Adding docs for pyproject.toml, and some additional instructions for trusted publishing via pypi.

### CODE REVIEW

File: docs/practices/customizing_project.rst [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+
+What is it? Why do it?
+-------------------------------------------------------------------------------
+
+The ``pyproject.toml`` file is used to define many different project configurations.
+It replaces the need for setup.py, requirements.txt, and other configuration files.
+
+A fundamental use for ``pyproject.toml`` is to define a dependencies. You should
+make an effort to add all the required dependencies to the ``[dependencies]``
+section of the file.
```
Comment (delucchi-cmu) [inline docs/practices/customizing_project.rst:12]:
```suggestion
A fundamental use for ``pyproject.toml`` is to define dependencies. You should
make an effort to add all the required dependencies to the ``[dependencies]``
section of the file. The template also creates a section for dependencies used only in development: ``[project.optional-dependencies][dev]``. You can add additional sets of optional dependencies here, e.g. for extended feature sets.
```

File: docs/practices/customizing_project.rst [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+In addition to dependencies, ``pyproject.toml`` centralizes the project's
+configuration and makes it easier to manage. If you examine your ``pyproject.toml``
+file, you'll see sections with titles like ``[tool.<blah>]``. These sections are
+used to configure tools that are used in the project that would otherwise require
+their own configuration files.
+
+A ``pyproject.toml`` file provides a vast number of customization options well
+beyond what can be covered here. For an in-depth look at how the file can be used
+and how it compares to the older ``setup.py`` checkout the
+`Python Packagining User Guide <https://packaging.python.org/en/latest/guides/writing-pyproject-toml/>`_.
```
Comment (delucchi-cmu) [inline docs/practices/customizing_project.rst:23]:
```suggestion
`Python Packaging User Guide <https://packaging.python.org/en/latest/guides/writing-pyproject-toml/>`_.
```

File: docs/practices/customizing_project.rst [hunk 3]
```diff
@@ -0,0 +1,23 @@
+Configuration with pyproject.toml
+===============================================================================
+
+What is it? Why do it?
+-------------------------------------------------------------------------------
+
+The ``pyproject.toml`` file is used to define many different project configurations.
+It replaces the need for setup.py, requirements.txt, and other configuration files.
+
+A fundamental use for ``pyproject.toml`` is to define a dependencies. You should
```
Comment (wilsonbb) [inline docs/practices/customizing_project.rst:10]: s/a dependencies/dependencies

**Review by wilsonbb — APPROVED** [review]
Just a small typo, LGTM!

File: docs/practices/customizing_project.rst (hunk 3 above)
Comment (drewoldag) [inline docs/practices/customizing_project.rst:10]:
```suggestion
A fundamental use for ``pyproject.toml`` is to define dependencies. You should
```

File: docs/practices/customizing_project.rst [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+
+What is it? Why do it?
+-------------------------------------------------------------------------------
+
+The ``pyproject.toml`` file is used to define many different project configurations.
+It replaces the need for setup.py, requirements.txt, and other configuration files.
+
+A fundamental use for ``pyproject.toml`` is to define dependencies. You should
+make an effort to add all the required dependencies to the ``[dependencies]``
+section of the file.
```
Comment (drewoldag) [inline docs/practices/customizing_project.rst:12]:
```suggestion
section of the file. The template also creates a section for dependencies used only in development: ``[project.optional-dependencies][dev]``. You can add additional sets of optional dependencies here, e.g. for extended feature sets.
```

**Pull request merged**